src/cli/config_validation.go

```diff
@@ -0,0 +1,227 @@
+/*
+* ZDNS Copyright 2024 Regents of the University of Michigan
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not
+* use this file except in compliance with the License. You may obtain a copy
+* of the License at http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+* implied. See the License for the specific language governing
+* permissions and limitations under the License.
+ */
+package cli
+
+import (
+	"fmt"
+	"net"
+	"net/netip"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/zmap/dns"
+
+	"github.com/zmap/zdns/src/internal/util"
+	"github.com/zmap/zdns/src/zdns"
+)
+
+const (
+	loopbackAddrSubnet = "127.0.0.0/8"
+	loopbackAddrString = "127.0.0.1"
+)
+
+func validateNetworkingConfig(gc *CLIConf) error {
+	// mutually exclusive CLI options
+	if gc.LocalIfaceString != "" && gc.LocalAddrString != "" {
+		return errors.New("--local-addr and --local-interface cannot both be specified")
+	}
+
+	// Note: we rely on the value of gc.UsingLoopbackNameServer set here, so this must be called first before other validation
+	if err := validateNameServers(gc); err != nil {
+		return errors.Wrap(err, "name servers did not pass validation")
+	}
+
+	if err := validateClientSubnetString(gc); err != nil {
+		return errors.Wrap(err, "client subnet did not pass validation")
+	}
+
+	if GC.LocalAddrString != "" {
```

Comment:
This was here before, but would this better be named SourceAddr? It really threw me off what LocalAddr meant when reviewing this.

Reply:
So I had the same thought, except it's called `localAddr` in the `net.Dialer` [pkg](https://pkg.go.dev/net#Dialer). I feel like for consistency we should keep it.